avoinspector/src/main/java/app/avo/inspector/AvoInspector.java

```diff
@@ -53,17 +53,20 @@ public AvoInspector(@NonNull String apiKey, @NonNull Application application, @N
 
     @SuppressWarnings("deprecation")
     public AvoInspector(@NonNull String apiKey, @NonNull Application application, @NonNull AvoInspectorEnv env, @Nullable Activity rootActivityForVisualInspector) {
-        String appVersionString = "";
         try {
             PackageManager packageManager = application.getPackageManager();
             PackageInfo pInfo = packageManager.getPackageInfo(application.getPackageName(), 0);
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-                appVersion = pInfo.getLongVersionCode();
-            } else {
-                //noinspection deprecation
-                appVersion = (long) pInfo.versionCode;
-            }
+
             appVersionString = pInfo.versionName;
+
+            if (!appVersionString.matches("^\\d+\\.\\d+\\.\\d+(?:-[\\w.-]+)?(?:\\+[\\w.-]+)?$")) {
```

Comment:
We have seen app versions that have something before the app version like

`MICROFRONTENT-1.2.3`
or `1.2.3+10`

Do we want to catch these as well?